makefiles/boot/riotboot.mk

```diff
@@ -74,18 +74,26 @@ $(BINDIR_APP)-slot1.hdr: OFFSET=$(SLOT1_IMAGE_OFFSET)
 # Generic target to create a binary files for both slots
 riotboot: $(SLOT_RIOT_BINS)
 
-# riotboot bootloader compile target
-riotboot/flash-bootloader: riotboot/bootloader/flash
+# riotboot required environment variables
 # IOTLAB_NODE is passed so that FLASHFILE is also set in the recursive make call
 # when PROGRAMMER=iotlab
+RIOTBOOT_BOOTLOADER_VARS = \
+    BOARD \
+    EXTERNAL_BOARD_DIRS \
+    DEBUG_ADAPTER_ID \
+    PROGRAMMER \
+    PROGRAMMER_QUIET \
+    IOTLAB_NODE \
+    #
+RIOTBOOT_BOOTLOADER_ENV_VARS := $(strip $(foreach varname,$(RIOTBOOT_BOOTLOADER_VARS), \
+    $(if $(filter file environment command,$(origin $(varname))), \
```

Comment:
I have a very low-key preference for this to be phrased positively ("repeat unless it's undefined, default or automatic" -- where would "override" go?), but please just act on this if that comment makes you go "yeah, that'd actually be better".

Reply:
The unchecked ones are indeed subtle behaviour changes, but if they were not overridden somewhere then the same default values will be used on the remote make call.

But PROGRAMMER_QUIET, should be set... I'll take a look

Reply:
My impression is that PROGRAMMER_QUIET defaults to QUIET, and thus its origin is default and it can rightfully be omitted.

Reply:
Seems its a matter of the order of inclusion of the Makefiles and that I'm using `:=` for the assignment, But now that I look at it quoting is not a good thing in all cases, since we are sometimes checking explictly like `ifeq (0,$(PROGRAMMER_QUIET))`, this change might not be worth it if handling all these cases, since I'll have to quote some and not others...

(thanks for taking a deeper look into this)